### ISSUE TITLE
Speed issue in inference in newer versions > 0.4.3 for Q/A

### ISSUE THREAD

**Issue by bipinkc19** [deepset-ai/FARM#486]
**Slow inference speed**
Inference speed is really slow in new versions > 0.4.3 gor Q/A

**Additional context/To reproduce**
Here is an example to reproduce the slow speed in inference

Collab notebook using farm==0.4.3
https://colab.research.google.com/drive/1ki4-HsRvVx62HVPRacx9lkwOESZ_o0gF?usp=sharing

The speed is very fine here

Collab notebook using farm==0.4.5
https://colab.research.google.com/drive/1JIWmbEwvGHuArWy3xsdr9z_srMx63fH-?usp=sharing

The speed is very slow here.
I have tried it even on the latest farm version using dedicated QAInferencer but the speed is still huge problem as in version 0.4.3 it is very fast
 

**System:**
Collab

**Comment by bipinkc19** [comment]
The reason for using the latest version is to get the context of the answers which is not provided in version 0.4.3

**Comment by Timoeller** [comment]
Hey @bipinkc19 thanks for the detailed report - we really appreciate issue reports that are being this reproducible.

That way we are able very fast to figure out the glitch. It is not the FARM version that slows it down but in the second example you do not have the GPU enable in the runtime. (The first notebook uses the GPU)

"INFO - farm.utils -   device: cpu n_gpu: 0, [...]"

**Comment by bipinkc19** [comment]
Thanks for the quick reply @Timoeller . But the notebook uses GPU. I have updated the notebook above with a code snippet ina  cell to check GPU availibility.

Looks like due to some error FARM isn't detecting and utilizing the GPU. Maybe it is related to:
 https://github.com/deepset-ai/FARM/issues/117

**Comment by Timoeller** [comment]
Nice, thanks for the additional input. Farm uses pytorch for utilizing the GPU - so I rather suspect it is a combination of colab + cuda + pytorch.

- I tried the notebook with FARM version 0.4.6 and coulnt make it work right away. 
- I have seen "Found existing installation: torch 1.6.0+cu101". 
- So I installed the correct pytorch version with `!pip install torch==1.5.1+cu101 torchvision==0.6.0+cu101 -f https://download.pytorch.org/whl/torch_stable.html` in combination with farm==0.4.6. There it works


```
Inferencing Samples: 100%|██████████| 16/16 [00:01<00:00,  8.02 Batches/s]CPU times: user 1.35 s, sys: 761 ms, total: 2.11 s
Wall time: 2.35 s
